src/main/java/com/google/cloud/genomics/dataflow/functions/AlleleSimilarityCalculator.java

```diff
@@ -58,38 +58,39 @@ public void processElement(ProcessContext context) {
     CallSimilarityCalculator callSimilarityCalculator =
         callSimilarityCalculatorFactory.get(isReferenceMajor(variant));
     FluentIterable<KV<Call, Call>> pairs =
-        PairGenerator.<Call, ImmutableList<Call>>allPairs(getSamplesWithVariant(variant));
+        PairGenerator.<Call, ImmutableList<Call>>allPairs(getSamplesWithVariant(variant), false,
+            new VariantUtils.CallComparator());
     for (KV<Call, Call> pair : pairs) {
       accumulateCallSimilarity(callSimilarityCalculator, pair.getKey(), pair.getValue());
     }
-    ++numberOfVariants;
   }
 
   private void accumulateCallSimilarity(CallSimilarityCalculator callSimilarityCalculator,
       Call call1, Call call2) {
     KV<String, String> callPair = KV.of(call1.getCallSetName(), call2.getCallSetName());
-    if (!accumulator.containsKey(callPair)) {
-      accumulator.put(callPair, 0.0);
+    KV<Double, Integer> callPairAccumulation = accumulator.get(callPair);
+    if (callPairAccumulation == null) {
+      callPairAccumulation = KV.of(0.0, 0);
+      accumulator.put(callPair, callPairAccumulation);
     }
-    accumulator.put(callPair,
-        accumulator.get(callPair) + callSimilarityCalculator.similarity(call1, call2));
+    accumulator.put(callPair, KV.of(
+        callPairAccumulation.getKey() + callSimilarityCalculator.similarity(call1, call2),
+        callPairAccumulation.getValue() + 1));
   }
 
   @Override
   public void finishBundle(Context context) {
-    for (KV<String, String> entry : accumulator.keySet()) {
-      String call1 = entry.getKey();
-      String call2 = entry.getValue();
-      KV<String, String> call12 = KV.of(call1, call2);
-      double sumOfRatios = accumulator.get(call12);
-      context.output(KV.of(call12, KV.of(sumOfRatios, numberOfVariants)));
+    Iterator<Entry<KV<String, String>, KV<Double, Integer>>> it = accumulator.entrySet().iterator();
```

Comment:
this should be a plain-jane for-each loop:

    for (Map.Entry<KV<String, String>, KV<Double, Integer>> entry : accumulator.entrySet()) {
      context.output(KV.of(entry.getKey(), entry.getValue()));
    }

Also, might want to consider extracting it out as a more general pattern since it doesn't really have anything to do with Strings:

    static <K, V> void output(DoFn<?, KV<K, V>>.Context context, Map<? extends K, ? extends V> map) {
      for (Map.Entry<? extends K, ? extends V> entry : map.entrySet()) {
        context.output(KV.<K, V>of(entry.getKey(), entry.getValue()));
      }
    }